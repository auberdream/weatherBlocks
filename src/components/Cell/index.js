import styled from 'styled-components';

export default styled.div`
    background-color: ${({ weather, theme }) => theme.palette.weather[weather]};
    width: 100%;
    padding: 2px;
`;
