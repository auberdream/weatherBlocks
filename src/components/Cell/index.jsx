import styled from 'styled-components';
import { shade } from 'polished';

export default styled.div`
    background-color: ${({ color, theme }) => theme.palette.weather[color]};
    padding: 2px;
    transition: background-color 300ms ease;
    width: 100%;

    &:hover {
        background-color: ${({ color, theme }) => shade(0.1, theme.palette.weather[color])};
    }
`;
