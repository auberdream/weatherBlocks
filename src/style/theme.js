export default {
  palette: {
    weather: {
      sunny: '#f72585',
      cloudy: '#b5179e',
      mist: '#7209b7',
      rain: '#560bad',
      thunder: '#3a0ca3',
      snow: '#3f37c9',
      sleet: '#4361ee',
      fog: '#4cc9f0',
    },
  },
  select: {
    background: '#FFF',
  },
};
