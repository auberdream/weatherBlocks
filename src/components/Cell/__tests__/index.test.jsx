import React from 'react';
import renderer from 'react-test-renderer';
import { Grommet } from 'grommet';

import theme from '../../../style/theme';
import Cell from '..';

it('Renders correctly for sunny', () => {
  const tree = renderer.create(
    <Grommet theme={theme}>
      <Cell color="sunny" />
    </Grommet>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly for cloudy', () => {
  const tree = renderer.create(
    <Grommet theme={theme}>
      <Cell color="cloudy" />
    </Grommet>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly for rain', () => {
  const tree = renderer.create(
    <Grommet theme={theme}>
      <Cell color="rain" />
    </Grommet>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly for snow', () => {
  const tree = renderer.create(
    <Grommet theme={theme}>
      <Cell color="snow" />
    </Grommet>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly for sleet', () => {
  const tree = renderer.create(
    <Grommet theme={theme}>
      <Cell color="sleet" />
    </Grommet>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly for thunder', () => {
  const tree = renderer.create(
    <Grommet theme={theme}>
      <Cell color="thunder" />
    </Grommet>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly for fog', () => {
  const tree = renderer.create(
    <Grommet theme={theme}>
      <Cell color="fog" />
    </Grommet>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly for mist', () => {
  const tree = renderer.create(
    <Grommet theme={theme}>
      <Cell color="mist" />
    </Grommet>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
