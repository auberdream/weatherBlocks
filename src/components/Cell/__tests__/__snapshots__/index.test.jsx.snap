// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly for cloudy 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 djOnwZ"
>
  <div
    className="sc-bdfBwQ jlnUAq"
    color="cloudy"
  />
</div>
`;

exports[`Renders correctly for fog 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 djOnwZ"
>
  <div
    className="sc-bdfBwQ cyFUSF"
    color="fog"
  />
</div>
`;

exports[`Renders correctly for mist 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 djOnwZ"
>
  <div
    className="sc-bdfBwQ cRmnOk"
    color="mist"
  />
</div>
`;

exports[`Renders correctly for rain 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 djOnwZ"
>
  <div
    className="sc-bdfBwQ jtWZpm"
    color="rain"
  />
</div>
`;

exports[`Renders correctly for sleet 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 djOnwZ"
>
  <div
    className="sc-bdfBwQ cWnJKE"
    color="sleet"
  />
</div>
`;

exports[`Renders correctly for snow 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 djOnwZ"
>
  <div
    className="sc-bdfBwQ eSAyLg"
    color="snow"
  />
</div>
`;

exports[`Renders correctly for sunny 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 djOnwZ"
>
  <div
    className="sc-bdfBwQ emqQwp"
    color="sunny"
  />
</div>
`;

exports[`Renders correctly for thunder 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 djOnwZ"
>
  <div
    className="sc-bdfBwQ bXzjgO"
    color="thunder"
  />
</div>
`;
